examples/ecs/event.rs

```diff
@@ -1,63 +1,144 @@
 }
 
-#[derive(Event)]
-struct MyEvent {
-    pub message: String,
-}
+// This event should be sent when an entity receives damage.
+#[derive(Event, Debug, Default)]
+struct DamageReceived;
 
-#[derive(Event, Default)]
-struct PlaySound;
+// This event should be sent when an entity blocks damage with armor.
+#[derive(Event, Debug, Default)]
+struct ArmorBlockedDamage;
 
-#[derive(Resource)]
-struct EventTriggerState {
-    event_timer: Timer,
-}
+// This resource represents a timer used to determine when to deal damage
+// By default it repeats once per second
+#[derive(Resource, Deref, DerefMut)]
+struct DamageTimer(pub Timer);
 
-impl Default for EventTriggerState {
+impl Default for DamageTimer {
     fn default() -> Self {
-        EventTriggerState {
-            event_timer: Timer::from_seconds(1.0, TimerMode::Repeating),
-        }
+        DamageTimer(Timer::from_seconds(1.0, TimerMode::Repeating))
     }
 }
 
-// sends MyEvent and PlaySound every second
-fn event_trigger(
+// Next we define systems that send, mutate, and receive events
+// This system reads 'DamageTimer', updates it, then sends a 'DealDamage' event
+// if the timer has finished.
+//
+// Events are sent using an 'EventWriter<T>' by calling 'send' or 'send_default'.
+// The 'send_default' method will send the event with the default value if the event
+// has a 'Default' implementation.
+fn deal_damage_over_time(
     time: Res<Time>,
-    mut state: ResMut<EventTriggerState>,
-    mut my_events: EventWriter<MyEvent>,
-    mut play_sound_events: EventWriter<PlaySound>,
+    mut state: ResMut<DamageTimer>,
+    mut events: EventWriter<DealDamage>,
 ) {
-    if state.event_timer.tick(time.delta()).finished() {
-        my_events.send(MyEvent {
-            message: "MyEvent just happened!".to_string(),
-        });
-        play_sound_events.send_default();
+    if state.tick(time.delta()).finished() {
+        // Events can be sent with 'send' and constructed just like any other object.
+        events.send(DealDamage { amount: 10 });
     }
 }
 
-// prints events as they come in
-fn event_listener(mut events: EventReader<MyEvent>) {
-    for my_event in events.read() {
-        info!("{}", my_event.message);
+// This system mutates the 'DealDamage' events to apply some armor value
+// It also sends an 'ArmorBlockedDamage' event if the value of 'DealDamage' is zero
+//
+// Events are mutated using an 'EventMutator<T>' by calling 'read'. This returns an iterator
+// over all the &mut T that this system has not read yet. Note, you can have multiple
+// 'EventReader', 'EventWriter', and 'EventMutator' in a given system, as long as the types (T) are different.
+fn apply_armor_to_damage(
+    mut dmg_events: EventMutator<DealDamage>,
+    mut armor_events: EventWriter<ArmorBlockedDamage>,
+) {
+    for event in dmg_events.read() {
+        event.amount -= 1;
+        if event.amount <= 0 {
+            // Events can be sent with 'send' event if they're zero sized
```

Comment:
what does this mean? It sounds like events must be zero sized to be sent

Reply:
typo it should say "Even if" but actually it doesn't add anything, I'll remove it.